app/main.py

```diff
@@ -1,2 +1,101 @@
+from typing import Any, Generator
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        self.max_capacity = 8
+        self.threshold = self.max_capacity - (self.max_capacity * 2 // 3)
+        self.hash_table = [None] * self.max_capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if self.hash_table.count(None) <= self.threshold:
+            self.resize_table()
+        index = self.find_index(key)
+        self.hash_table[index] = (key, value)
+
+    def __getitem__(self, key: Any) -> Any:
+        index = self.find_value_from_key(key)
+        if self.hash_table[index]:
+            return self.hash_table[index][1]
+        else:
+            raise KeyError("Key not found")
+
+    def __len__(self) -> int:
+        return len(self.hash_table) - self.hash_table.count(None)
+
+    def __delitem__(self, key: Any) -> None:
+        index = self.find_value_from_key(key)
+        if self.hash_table[index]:
+            del self.hash_table[index]
+        else:
+            raise KeyError("Key not found")
+
+    def find_index(self, key: Any) -> int:
```

Comment:
Both methods have almost identical probing logic. This can be simplified and unified into a single helper method.